Lib/asyncio/futures.py

```diff
@@ -132,6 +132,11 @@ def _make_cancelled_error(self):
         This should only be called once when handling a cancellation since
         it erases the saved context exception value.
         """
+        if isinstance(self._cancelled_exc, exceptions.CancelledError):
```

Comment:
The way I read the code in tasks.py, it will always be a CancelledError (or None).

I would also update the comment in tasks.py where it is set, because now we never chain it -- we just raise it in a different context.

Reply:
Agree, `_cancelled_exc` is always `CancelledError` by design. 
I've simplified the source code.